### PR TITLE
Fix array copying check.

### PR DESCRIPTION
**Pull request by chriseth**

### CODE REVIEW

File: libsolidity/codegen/YulUtilFunctions.cpp [hunk 1]
```diff
@@ -2416,7 +2416,7 @@ string YulUtilFunctions::copyArrayFromStorageToMemoryFunction(ArrayType const& _
 	return m_functionCollector.createFunction(functionName, [&]() {
 		if (_from.baseType()->isValueType())
 		{
-			solAssert(_from.baseType() == _to.baseType(), "");
+			solAssert(*_from.baseType() == *_to.baseType(), "");
```
Comment (hrkrshnn) [inline libsolidity/codegen/YulUtilFunctions.cpp:2419]: Can this assert be reproduced outside of using user defined value type?

Comment (chriseth) [reply libsolidity/codegen/YulUtilFunctions.cpp:2419]: Maybe for enums, but I didn't check. My change only relaxed the check, so I think it's fine.

**Review by hrkrshnn — DISMISSED** [review]

**Comment by chriseth** [comment]
Updated (forced abi v2)

**Review by hrkrshnn — APPROVED** [review]

**Comment by hrkrshnn** [comment]
What happens for abiv1?

**Comment by chriseth** [comment]
v1 cannot handle the struct.

**Pull request merged**